source/mainnet/smart-contracts/tutorials/piggy-bank/deploying.rst

```diff
@@ -0,0 +1,212 @@
+
+If everything works correctly the output has a green line as follows:
+
+.. code-block:: console
+
+   Successfully updated contract instance {"index":<smart-contract-instance-index>,"subindex":0} ('piggy_bank_part2_instance') using the function 'insert'.
+
+.. image:: ./images/pb_tutorial_23.png
+   :width: 100 %
+
+You can find additional information about updating a smart contract instance in this :ref:`guide<interact-instance>`.
+
+
+Viewing the piggy bank state
+============================
+
+You can display the raw return value of your piggy bank smart contract with the following command.
+
+.. code-block:: console
+
+   $./concordium-client contract invoke piggy_bank_part2_instance --entrypoint view --energy 1000 --grpc-port 10001
+
+If everything works correctly the output has a green line as follows:
+
+.. code-block:: console
+
+   Invocation resulted in success:
+   - Energy used: 510 NRG
+   - Return value (raw):
+   [0,64,66,15,0,0,0,0,0]
+   .
+
+.. image:: ./images/pb_tutorial_24.png
+   :width: 100 %
+
+You can find additional information about invoking in this :ref:`guide<invoke-instance>`.
+
+
+Smashing the piggy bank smart contract
+======================================
+
+Smash the piggy bank contract to test that you can invoke this entry point. After smashing the piggy bank you cannot enter any additional CCD into the piggy bank anymore.
+
+.. code-block:: console
+
+   $./concordium-client contract update piggy_bank_part2_instance --entrypoint smash --sender <account-name> --energy 2000 --grpc-port 10001
+
+
+If everything works correctly the output has a green line as follows:
+
+.. code-block:: console
+
+   Successfully updated contract instance {"index":<smart-contract-instance-index>,"subindex":0} ('piggy_bank_part2_instance') using the function 'smash'.
+
+.. image:: ./images/pb_tutorial_25.png
+   :width: 100 %
+
+You can find additional information about updating a smart contract instance in this :ref:`guide<interact-instance>`.
+
+.. note::
+
+   Ensure that you don't send any CCD to the smart contract instance and that the ``--sender`` flag uses the <account-name>
+   that initialized the piggy bank smart contract instance (the owner of this smart contract instance).
+   You can find explanations about the behavior of the ``Smash`` entry point in this :ref:`guide<smashing-the-piggy-bank-writing>`.
+   The ``Smash`` entry point is ``non-payable`` and will print the below error message if you send CCD with the ``--amount`` flag.
+   The opposite is not true and you can omit the ``--amount`` flag when interacting
+   with a ``payable`` function without raising an error because in that case 0 CCD is sent to the ``payable`` entry point.
+
+.. code-block:: console
+
+   Error: Updating contract instance failed:
+   'smash' in 'PiggyBank' at {"index":<smart-contract-instance-index>,"subindex":0} failed with code -2147483636.
```

Comment:
Please write that they can find the meaning of common error codes by looking here: https://docs.rs/concordium-std/3.0.0/concordium_std/#signalling-errors

(The errors from concordium-std start from `i32::MIN` and go upwards. The contract's own error types start from `-1` and go downwards.)